mm2src/adex_cli/src/scenarios/init_mm2_cfg.rs

```diff
@@ -0,0 +1,302 @@
+    pub userhome: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub dbdir: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub rpcip: Option<Ipv4Addr>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub rpcport: Option<u16>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub rpc_local_only: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub i_am_seed: Option<bool>,
+    #[serde(skip_serializing_if = "Vec::<Ipv4Addr>::is_empty")]
+    pub seednodes: Vec<Ipv4Addr>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub hd_account_id: Option<u64>,
+}
+
+impl Mm2Cfg {
+    pub fn new() -> Mm2Cfg {
+        Mm2Cfg {
+            gui: None,
+            net_id: None,
+            rpc_password: None,
+            passphrase: None,
+            allow_weak_password: None,
+            userhome: None,
+            dbdir: None,
+            rpcip: None,
+            rpcport: None,
+            rpc_local_only: None,
+            i_am_seed: None,
+            seednodes: Vec::<Ipv4Addr>::new(),
+            hd_account_id: None,
+        }
+    }
+
+    fn inquire(&mut self) -> Result<(), ()> {
+        self.inquire_gui()?;
+        self.inquire_net_id()?;
+        self.inquire_passphrase()?;
+        self.inquire_allow_weak_password()?;
+        self.inquire_rpc_password()?;
+        self.inquire_userhome()?;
+        self.inquire_dbdir()?;
+        self.inquire_rpcip()?;
+        self.inquire_rpcport()?;
+        self.inquire_rpc_local_only()?;
+        self.inquire_i_am_a_seed()?;
+        self.inquire_seednodes()?;
+        self.inquire_hd_account_id()?;
+        Ok(())
+    }
+
+    fn inquire_dbdir(&mut self) -> Result<(), ()> {
+        let is_reachable_dir = |dbdir: &InquireOption<String>| -> Result<Validation, CustomUserError> {
+            match dbdir {
+                InquireOption::None => Ok(Validation::Valid),
+                InquireOption::Some(dbdir) => {
+                    let path = Path::new(dbdir);
+                    if path.is_dir().not() {
+                        return Ok(Validation::Invalid(
+                            format!("\"{dbdir}\" - is not a directory or does not exist").into(),
+                        ));
+                    }
+                    Ok(Validation::Valid)
+                },
+            }
+        };
+
+        self.dbdir = CustomType::<InquireOption<String>>::new("What is dbdir")
+                .with_placeholder(DEFAULT_OPTION_PLACEHOLDER)
+                .with_help_message("AtomicDEX API database path. Optional, defaults to a subfolder named DB in the path of your mm2 binary")
+                .with_validator(is_reachable_dir)
+                .prompt()
+                .map_err(|error| {
+                    error!("Failed to get dbdir: {error}");
+                })?.into();
+
+        Ok(())
+    }
+
+    fn inquire_gui(&mut self) -> Result<(), ()> {
```

Comment:
might be useful to keep it since it would allow us to track "CLI users" - we could, for instance, use the tag "adex-cli" for the `gui` field by default, when using ADEX API via `adex-cli`.

Reply:
done

Reply:
> for instance, use the tag "adex-cli" for the gui field by default, when using ADEX API via adex-cli

I agree, having a default value for this and not allowing the user to set it seems the right way to do this.

Reply:
done